pkg/asset/machines/azure/azuremachines.go

```diff
@@ -58,7 +58,8 @@ func GenerateMachines(platform *azure.Platform, pool *types.MachinePool, userDat
 					Offer:     osImage.Offer,
 					SKU:       osImage.SKU,
 				},
-				Version: osImage.Version,
+				Version:         osImage.Version,
+				ThirdPartyImage: true,
```

Comment:
This can default to true, but should take into account the `imagePurchasePlan` (ug Azure API design):

https://github.com/openshift/installer/blob/acd8593004f6fd24785b7083639ee5a922ceee69/pkg/types/azure/machinepool.go#L199-L202

So if plan is unset or set to `WithPurchasePlan` then this would be true. But if set to `NoPurchasePlan` it would be false.